src/components/EditTextModal/EditTextModal.js

```diff
@@ -0,0 +1,160 @@
+  };
+
+  useEffect(
+    () => {
+      if (isOpen) {
+        //  prepopulate selected text
+        setNewText(core.getSelectedText());
+        // convert and store selected text coordinates
+        storeConvertedPDFCoordinates();
+      }
+    },
+    [isOpen]
+  );
+
+  // core.getSelectedTextQuads() returns 4 coordinates pair
+  // (x4,y4)---(x3,y3)
+  //   |          |
+  // (x1,y1)---(x2,y2)
+  // Firstly convert it to PDF coordinates and then store it in a flat structure
+  const storeConvertedPDFCoordinates = () => {
+    const textCoordinates = core.getSelectedTextQuads();
+
+    // Example:
+    // textCoordinates = {
+    //   1: [
+    //     {x1: 32, y1: 582, x2: 67.28, y2: 582, x3: 67.28,4: 32, y1: 582, y2: 582, y3: 565, y4: 565},
+    //     {x1: 32, x2: 68, x3: 68, x4: 32, y1: 595, y2: 595, y3: 579, y4: 579},
+    //   ]
+    // }
+    // convert to a flat structure
+    // textCoordinates = [{x1:32, x2:1, y1:582, y2:2, pageNum:1}, {x1:1, x2:1, y1:2, y2:2, pageNum:1}]
+
+    if (textCoordinates) {
+      const convertedTextCoordinates = [];
+      Object.keys(textCoordinates).forEach(pageNum => {
+        textCoordinates[pageNum].forEach(rect => {
+          const point1 = window.docViewer.getDocument().getPDFCoordinates(pageNum, rect.x1, rect.y1);
+          const point2 = window.docViewer.getDocument().getPDFCoordinates(pageNum, rect.x3, rect.y3);
+          convertedTextCoordinates.push({ x1: point1.x, x2: point2.x, y1: point1.y, y2: point2.y,pageNum:Number(pageNum) });
+        });
+      });
+      setTextCoordinates(convertedTextCoordinates);
+    }
+  };
+
+  const updateText = e => {
+    e.preventDefault();
+    window.PDFNet.runWithCleanup(async() => {
+      const doc = await window.docViewer.getDocument().getPDFDoc();
+      // Treat each line as a rectangle region and update them separately
+      const lines = newText.split(/\r?\n/);
+      doc.initSecurityHandler();
+      doc.lock();
+      const PDFNet = window.PDFNet;
+      // Note: If the number of lines are more than number of rects defined in textCoordinates, then we just ignore the rest
+      if (textCoordinates && textCoordinates.length > 0) {
+        for (let i=0; i<textCoordinates.length; i++) {
+          const { x1, x2, y1, y2, pageNum } = textCoordinates[i];
+          const [page, replacer, region] = await Promise.all([
+            doc.getPage(pageNum),
+            PDFNet.ContentReplacer.create(),
+            PDFNet.Rect.init(x1, y1, x2, y2)
+          ]);
+          replacer.addText(region, lines[i]);
+          replacer.process(page);
+          replacer.destroy();
+        }
+
+        // clear the cache
+        window.docViewer.refreshAll();
+        // update viewer with new document
+        window.docViewer.updateView();
+        // Annotations may contain text so we need to regenerate
+        // our text representation
+        window.docViewer.getDocument().refreshTextData();
+      }
+      // doc.unlock();
```

Comment:
For some reason, this unlock gives me a warning:
`[WARNING], the object to be unlocked was not found in the unlock list. Unlocking may cause errors.`

I am not sure what's wrong with it

Reply:
I think runWithCleanup will automatically unlock for you https://www.pdftron.com/documentation/web/guides/full-api/advanced-features/#full-api-locking

So you can just delete that line